target_chains/ethereum/sdk/solidity/MockPyth.sol

```diff
@@ -160,6 +161,121 @@ contract MockPyth is AbstractPyth {
             );
     }
 
+    function parseTwapPriceFeedUpdates(
+        bytes[] calldata updateData,
+        bytes32[] calldata priceIds
+    )
+        external
+        payable
+        override
+        returns (PythStructs.TwapPriceFeed[] memory twapPriceFeeds)
+    {
+        // Validate inputs and fee
+        if (updateData.length != 2) revert PythErrors.InvalidUpdateData();
+
+        uint requiredFee = getUpdateFee(updateData);
+        if (msg.value < requiredFee) revert PythErrors.InsufficientFee();
+
+        twapPriceFeeds = new PythStructs.TwapPriceFeed[](priceIds.length);
+
+        // Process each price ID
+        for (uint i = 0; i < priceIds.length; i++) {
+            processTwapPriceFeed(updateData, priceIds[i], i, twapPriceFeeds);
+        }
+
+        return twapPriceFeeds;
+    }
+
+    function findPriceFeed(
+        bytes[] calldata updateData,
+        bytes32 priceId,
+        uint index
+    )
+        private
+        pure
+        returns (
+            PythStructs.PriceFeed memory feed,
+            uint64 prevPublishTime,
+            bool found
+        )
+    {
+        (feed, prevPublishTime) = abi.decode(
+            updateData[index],
+            (PythStructs.PriceFeed, uint64)
+        );
+
+        if (feed.id == priceId) {
+            found = true;
+        }
+    }
+
+    function processTwapPriceFeed(
+        bytes[] calldata updateData,
+        bytes32 priceId,
+        uint index,
+        PythStructs.TwapPriceFeed[] memory twapPriceFeeds
+    ) private {
+        // Decode start and end TWAP info
+        (bytes32 startId, PythStructs.TwapPriceInfo memory startInfo) = abi
+            .decode(updateData[0], (bytes32, PythStructs.TwapPriceInfo));
+        (bytes32 endId, PythStructs.TwapPriceInfo memory endInfo) = abi.decode(
+            updateData[1],
+            (bytes32, PythStructs.TwapPriceInfo)
+        );
+
+        // Validate IDs match
+        if (startId != priceId || endId != priceId)
+            revert PythErrors.InvalidTwapUpdateDataSet();
+
+        // Validate time ordering
+        if (startInfo.publishTime >= endInfo.publishTime) {
+            revert PythErrors.InvalidTwapUpdateDataSet();
+        }
+
+        if (startInfo.publishSlot >= endInfo.publishSlot) {
+            revert PythErrors.InvalidTwapUpdateDataSet();
+        }
+
+        // Calculate TWAP
+        twapPriceFeeds[index] = PythUtils.calculateTwap(
+            priceId,
+            startInfo,
+            endInfo
+        );
+
+        // Emit event
+        emit TwapPriceFeedUpdate(
+            priceId,
+            startInfo.publishTime,
+            endInfo.publishTime,
+            twapPriceFeeds[index].twap.price,
+            twapPriceFeeds[index].twap.conf,
+            twapPriceFeeds[index].downSlotsRatio
+        );
+    }
+
+    function createTwapPriceFeedUpdateData(
```

Comment:
I like you to replace it with a function that creates `TwapPriceFeed` and doesn't take cumulative/... ; those things are internal things that external people don't know.

as an external developer i want to cal createTwapPriceFeedUpdateData() a couple of times and make sure putting it as the input to parseTwap gives me the result. This will make your code much simpler.